From a6b4c1f26971b3428e437423459fbe344e831a99 Mon Sep 17 00:00:00 2001
From: Bassel Saba <basselsa@marvell.com>
Date: Sun, 11 Oct 2015 17:16:07 +0300
Subject: [PATCH 054/538] fix: sdhci-pxav3: Disable clock inversion for HS SD
 cards

The Clock inversion option is needed to support slow frequencies
when the card input hold time requirement is high.
This setting is not required for high speed SD and might cause timing
violation.
Erratum 'FE-2946959' was recently updated to specify this HS SD setting

Change-Id: I4d91f0707e79cf1fb4a392ea9424cbf647a8f6ed
Signed-off-by: Bassel Saba <basselsa@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/23934
Tested-by: Star_Automation <star@marvell.com>
Reviewed-by: Nadav Haklai <nadavh@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/27302
Tested-by: Lior Amsalem <alior@marvell.com>
---
 drivers/mmc/host/sdhci-pxav3.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/mmc/host/sdhci-pxav3.c b/drivers/mmc/host/sdhci-pxav3.c
index 8742afd..22a0176 100644
--- a/drivers/mmc/host/sdhci-pxav3.c
+++ b/drivers/mmc/host/sdhci-pxav3.c
@@ -345,7 +345,8 @@ static void pxav3_set_uhs_signaling(struct sdhci_host *host, unsigned int uhs)
 		    uhs == MMC_TIMING_UHS_DDR50) {
 			reg_val &= ~SDIO3_CONF_CLK_INV;
 			reg_val |= SDIO3_CONF_SD_FB_CLK;
-		} else if (uhs == MMC_TIMING_MMC_HS) {
+		} else if (uhs == MMC_TIMING_MMC_HS ||
+			   uhs == MMC_TIMING_SD_HS) {
 			reg_val &= ~SDIO3_CONF_CLK_INV;
 			reg_val &= ~SDIO3_CONF_SD_FB_CLK;
 		} else {
-- 
1.9.1

